packages/ckeditor5-dev-env/lib/release-tools/utils/transformcommitforckeditor5package.js

```diff
@@ -0,0 +1,141 @@
+ *   - filters out the commit if it should not be visible in the changelog,
+ *   - makes links to issues and user's profiles on GitHub.
+ *
+ * @param {Commit} commit
+ * @param {Boolean} [displayLog=true] Whether to display the logs.
+ * @returns {Commit}
+ */
+module.exports = function transformCommitForCkeditor5Package( commit, displayLog = true ) {
+	const log = logger( displayLog ? 'info' : 'error' );
+
+	// For merge commit from Github, additional description is provided as "footer".
+	if ( !commit.body && commit.footer ) {
+		commit.body = commit.footer;
+		commit.footer = null;
+	}
+
+	if ( commit.header.startsWith( 'Merge' ) ) {
+		const parsedHeader = parserOptions.headerPattern.exec( commit.body );
+
+		if ( parsedHeader ) {
+			parserOptions.headerCorrespondence.forEach( ( key, index ) => {
+				commit[ key ] = parsedHeader[ index + 1 ];
+			} );
+
+			// Remove the new header from commit body in order to avoid
+			// duplicating the same sentence in a changelog description.
+			commit.body = commit.body.replace( parserOptions.headerPattern, '' ).trim();
+		}
+	}
+
+	if ( typeof commit.hash === 'string' ) {
+		commit.hash = commit.hash.substring( 0, 7 );
+	}
+
+	const hasCorrectType = transformCommit.availableTypes.has( commit.type );
+	const isCommitIncluded = transformCommit.availableTypes.get( commit.type );
+
+	let logMessage = `* ${ commit.hash } "${ commit.header }" `;
+
+	if ( hasCorrectType && isCommitIncluded ) {
+		logMessage += chalk.green( 'INCLUDED' );
+	} else if ( hasCorrectType && !isCommitIncluded ) {
+		logMessage += chalk.grey( 'SKIPPED' );
+	} else {
+		logMessage += chalk.red( 'INVALID' );
+	}
+
+	if ( commit.header.startsWith( 'Merge' ) && hasCorrectType ) {
+		const indentSize = '[XX:YY:ZZ] * 1234567 '.length;
+		logMessage += `\n${ ' '.repeat( indentSize ) }"${ commit.type }: ${ commit.subject }"`;
+	}
+
+	log.info( logMessage );
+
+	if ( !isCommitIncluded ) {
+		return;
+	}
+
+	const issues = [];
+
+	commit.rawType = commit.type;
+	commit.type = transformCommit.getCommitType( commit.type );
+
+	if ( typeof commit.subject === 'string' ) {
+		commit.subject = transformCommit.linkGithubIssues(
+			transformCommit.linkGithubUsers( commit.subject ), issues
+		);
+	}
+
+	if ( typeof commit.body === 'string' ) {
+		commit.body = commit.body.split( '\n' )
+			.map( ( line ) => {
+				if ( !line.length ) {
+					return '';
+				}
+
+				return '  ' + line;
+			} )
+			.join( '\n' );
+	}
+
+	for ( const note of commit.notes ) {
+		if ( note.title === 'BREAKING CHANGE' ) {
+			note.title = 'BREAKING CHANGES';
+		}
+		note.text = transformCommit.linkGithubIssues( transformCommit.linkGithubUsers( note.text ) );
+	}
+
+	// Removes references that already appear in the subject.
+	commit.references = commit.references.filter( ( reference ) => {
+		return issues.indexOf( reference.issue ) === -1;
+	} );
+
+	return commit;
+};
+
+/**
+ * @typedef {Object} Commit
+ *
+ * @property {String} [type] Type of the commit.
```

Comment:
Yes, if commit does not match to our convention, these fields won't have a value.